src/oop/object_interactions/medium/q2/MainTest.java

```diff
@@ -0,0 +1,207 @@
+                {"Sam", "Sam always cheat on exams", new String[]{"Bob", "Ali"}},
+                {"John", "John is in love with Alice", new String[]{"Alice", "Bob"}},
+                {"Kim", "Kim has broken her leg", new String[]{"Ali", "John"}},
+        };
+        for (int num = 0; num < test.length; num++) {
+            String name = (String) test[num][0];
+            String secret = (String) test[num][1];
+            String[] friends = (String[]) test[num][2];
+            Object[][] arguments = {
+                    {name, String.class},
+                    {secret, String.class},
+                    {friends, String[].class}
+            };
+            Object classInstance = testClass.createInstance(arguments);
+            _assertEquals(name, testClass.getFieldValue(classInstance, myNameAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertEquals(secret, testClass.getFieldValue(classInstance, mySecretAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertArrayEquals(friends, testClass.getFieldValue(classInstance, friendNamesAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            _assertArrayEquals(new String[friends.length], testClass.getFieldValue(classInstance, friendSecretsAttributeName),
+                    "You have modified the provided constructor on the " + className + " class. Please revert it to the original state.");
+            assertTrue(testClass.hasMethod(getNameMethodName, null, String.class, new String[]{"public"}),
+                    String.format("You have modified the provided %s method method. Please revert it to the original state.", getNameMethodName));
+            Object getMethodOutput = testClass.callMethod(getNameMethodName, new String[]{"public"}, classInstance);
+            _assertEquals(name, getMethodOutput,
+                    "You have modified the provided " + getNameMethodName + " method on the " + className + " class. Please revert it to the original state.");
+            assertTrue(testClass.hasMethod(getSecretMethodName, null, String.class, new String[]{"public"}),
+                    String.format("You have modified the provided %s method method. Please revert it to the original state.", getSecretMethodName));
+            getMethodOutput = testClass.callMethod(getSecretMethodName, new String[]{"public"}, classInstance);
+            _assertEquals(secret, getMethodOutput,
+                    "You have modified the provided " + getSecretMethodName + " method on the " + className + " class. Please revert it to the original state.");
+        }
+
+    }
+
+    private static Stream<Arguments> hearSecretInputProvider() {
```

Comment:
You should also make sure to have test cases where there is no match, and a test case with a longer list of names as well.